### PR TITLE
BE: Invite Model

### PR DESCRIPTION
**Pull request by robertjhull**
### What this PR does (required):
- Adds Invite model to back end

### Screenshots / Videos (required):
n/a

### Any information needed to test this feature (required):
n/a

### Any issues with the current functionality (optional):
n/a

### CODE REVIEW

File: server/models/Invite.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  },
+},
+{
+  timestamps: true
+}
+);
+
+inviteSchema.pre('validate', async function(next) {
+  // ensures one field exists but not both
+  return (!this.email !== !this.user)
```
Comment (ajaytomgeorge) [inline server/models/Invite.js:37]: user or userId?

**Review by ajaytomgeorge — COMMENTED** [review]
Looks good from my side

File: server/models/Invite.js (hunk 1 above)
Comment (robertjhull) [reply server/models/Invite.js]: Should have been userId - fixed.

File: server/models/Invite.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User',
+    required: true,
+  },
+  userId: {
+    type: String,
+    index: {
+      unique: true,
+      partialFilterExpression: {email: {$type: "string"}}
+    }
```
Comment (moffatethan) [inline server/models/Invite.js:14]: Nice work!

**Review by moffatethan — APPROVED** [review]
Good work Robert!

**Pull request merged**